NETCORE/src/Shared/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -332,7 +312,7 @@ private static void AddCommonInitializers(IServiceCollection services)
         private static void AddCommonTelemetryModules(IServiceCollection services)
         {
             // Previously users were encouraged to manually add the DiagnosticsTelemetryModule.
-            services.AddSingletonIfNotExists<ITelemetryModule, DiagnosticsTelemetryModule>();
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<ITelemetryModule, DiagnosticsTelemetryModule>());
```

Comment:
@TimothyMothra FYI I didn't make any other changes here but really all of the `services.AddSingleton<ITelemetryModule, ...>();` calls should be `TryAddEnumerable` calls. If user only calls registration once everything will be fine. But if they call one of the "Add" extensions more than once, they'll get duplicates of all of these modules. Typically "Add" extensions should be safe to be called more than once which is accomplished by calling `TryAddSingleton` \ `TryAddEnumerable` 🤷

Reply:
I don't remember the full history, but I reviewed all my previous commit comments.

It looks like I was trying to add safety checks for every possible way that a user could have added the DiagnosticsTelemetryModule ([link 1](https://github.com/microsoft/ApplicationInsights-dotnet/pull/1886/commits/8a512f7323c28957003f746a9fe5fd19c40de205))([link 2](https://github.com/microsoft/ApplicationInsights-dotnet/pull/1886/commits/058008653cd81d6f3e9bd2401abba97ed8f1fcb1)). The TryAdd is substantially better :)

Thank you